components/Form/AddressLookup/AddressLookup.tsx

```diff
@@ -95,27 +99,64 @@ const AddressLookup = ({
   const [postcode, setPostcode] = useState(
     defaultValue && defaultValue.postcode
   );
+  const [buildingNumber, setBuildingNumber] = useState(
+    defaultValue && defaultValue.buildingNumber !== undefined
+      ? defaultValue.buildingNumber
+      : ''
+  );
   const [results, setResults] = useState<Address[]>([]);
   const [isManually, setIsManually] = useState<boolean>();
   const [error, setError] = useState<string>();
+  const [loading, setLoading] = useState(false);
+
   const searchPostcode = useCallback(async () => {
     control.setValue(`address`, null);
     if (!postcode || !isPostcodeValid(postcode)) {
       setError('You entered an invalid postcode.');
       return;
     }
+    if (
+      buildingNumber?.length > 0 &&
+      buildingNumber !== '' &&
+      !isNumeric(buildingNumber)
+    ) {
+      setError('Building number must use valid characters (0-9)');
+      return;
+    }
+
+    setLoading(true);
     setIsManually(false);
     setError(undefined);
     setResults([]);
-    try {
-      const res = await lookupPostcode(postcode);
-      res.length === 0
-        ? setError('There was a problem with the postcode.')
-        : setResults(res);
-    } catch {
-      setError('There was a problem with the postcode.');
+    let page_number = 1;
+    let isLastPage = false;
+    const matchingAddresses: Address[] = [];
+    let errorMessage;
+
+    do {
+      try {
+        const { address, page_count } = await lookupPostcode(
+          postcode,
+          page_number,
+          buildingNumber
+        );
+
+        address.length === 0
+          ? (errorMessage = 'There was a problem with the postcode.')
+          : matchingAddresses?.push(...address);
+        page_number === page_count ? (isLastPage = true) : page_number++;
+      } catch {
+        errorMessage = 'There was a problem with the postcode.';
+      }
+    } while (!errorMessage && !isLastPage);
```

Comment:
Not 100% sure this is the most performant way of doing what we need, possibly making use of [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) might be what we need but this is good enough for now! Make it work, make it right then make it fast. Amirite?!

Reply:
oh, interesting, I hadn't thought of a Promise.all - I'll leave it as is for now but might revisit it with the Promise.all in mind